crates/base/src/index.rs

```diff
@@ -78,6 +78,41 @@ pub enum StatError {
     NotExist,
 }
 
+#[must_use]
+#[derive(Debug, Clone, Error, Serialize, Deserialize)]
+pub enum SettingError {
+    #[error("Setting key {key} is not exist.")]
+    BadKey { key: String },
+    #[error("Setting key {key} has a wrong value {value}.")]
+    BadValue { key: String, value: String },
+    #[error("Index not found.")]
+    NotExist,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct IndexFlexibleOptions {
+    #[serde(default = "IndexFlexibleOptions::default_optimizing_threads")]
+    pub optimizing_threads: u16,
+}
+
+impl IndexFlexibleOptions {
+    pub fn default_optimizing_threads() -> u16 {
+        match std::thread::available_parallelism() {
+            Ok(threads) => (threads.get() as f64).sqrt() as _,
+            Err(_) => 1,
```

Comment:
We may set it to `1` in the future... I realized that sqrt of number of cores is a little stupid: I'm too lazy to set this value when testing indexes so such a number came up...